src/irr/ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -179,42 +179,58 @@ bsdf_data_t irr_glsl_MC_fetchBSDFData(in uint ix)
 _IRR_STATIC_INLINE_CONSTEXPR const char* FRAGMENT_SHADER_IMPL = R"(
 #include <irr/builtin/glsl/format/decode.glsl>
 
-instr_stream_t getEvalStream()
+instr_stream_t getEvalStream(in MC_precomputed_t precomp)
 {
-	instr_stream_t stream;
-	stream.offset = InstData.data[InstanceIndex].instr_offset;
-	stream.count = InstData.data[InstanceIndex].rem_pdf_count;
+	instr_stream_t front;
+	front.offset = InstData.data[InstanceIndex].front_instr_offset;
+	front.count = InstData.data[InstanceIndex].front_rem_pdf_count;
 
-	return stream;
+	instr_stream_t back;
+	back.offset = InstData.data[InstanceIndex].back_instr_offset;
+	back.count = InstData.data[InstanceIndex].back_rem_pdf_count;
+
+	return precomp.NdotV > 0.0 ? front : back;
```

Comment:
use a proper branch... otherwise you're doing 2x the memory fetches

also its not `NdotV` that you must use (interpolated shading normals can fuck this up) .... but take in a boolean `frontFace` which in the rasterization shader would be `gl_FrontFacing`